### PR TITLE
 Integrate MALDIquant's baseline estimation #119 

### PR DESCRIPTION
**Pull request by lgatto**
Addresses #119

### CODE REVIEW

**Comment by lgatto** [comment]
The GHA fails - I have tried to fiddle with it, so just ignore the workflow file for now. (Or any advise welcome)

**Comment by lgatto** [comment]
General points to consider for this review:
1. Example data used in `?estimateBaseline` comes from MALDIquant, which is thus now in suggest, rather than part of the package itself (size is about 0.5 Mb)
2. There's one single `estimateBaseline()` function with a method argument. We could also provide individual `estimateBaselineSNIP()`, `estimateBaselineTopHat()`, `esimateBaselineConvexHull()` and `estimateBaselineMedian()`.

**Comment by lgatto** [comment]
In addition to using [simulated example data](https://github.com/rformassspectrometry/MsCoreUtils/issues/119#issuecomment-1870249152), I also export the individual `estimateBaseline[Snip|TopHat|ConvexHub|Media]()` functions. I think this makes arguments explicit, and clarifies the manual page. The main `estimateBaseline()` function is still available, so as to remain backward compatible with the signature in `MALDIquant`.

File: R/estimateBaseline.R [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+##' ## Median baseline
+##' base_med <- estimateBaseline(mz, intensity,
+##'                              method = "median")
+##' ## same as estimateBaselineMedian(mz, intensity)
+##' lines(mz, base_med, col = "orange")
+##'
+##' legend("topright", lwd = 1,
+##'         legend = c("SNIP", "TopHat (hws = 75)",
+##'                    "TopHat (hws = 150)",
+##'                    "ConvexHull", "Media"),
```
Comment (sgibb) [inline R/estimateBaseline.R:153]:
```suggestion
##'                    "ConvexHull", "Median"),
```

File: R/estimateBaseline.R [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+##' ## add noise
+##' intensity <- intensity + rnorm(nmz, mean = 0, sd = 1)
+##'
+##' plot(mz, intensity, type = "l")
+##'
+##' ## ----------------------------
+##' ## SNIP baseline
+##' base_SNIP <- estimateBaseline(mz, intensity,
+##'                               method = "SNIP",
+##'                               iterations = 100L)
```
Comment (sgibb) [inline R/estimateBaseline.R:119]:
```suggestion
##'                               iterations = 20L)
```

I just adapted the iterations/hws to fit better to the example data.

File: R/estimateBaseline.R [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+##' base_TH75 <- estimateBaseline(mz, intensity,
+##'                               method = "TopHat",
+##'                               halfWindowSize = 75L)
+##' ## same as estimateBaselineTopHat(mz, intenstity, halfWindowSize = 75L)
+##' lines(mz, base_TH75, col = "blue")
+##'
+##' base_TH150 <- estimateBaseline(mz, intensity,
+##'                                method = "TopHat",
+##'                                halfWindowSize = 150L)
+##' lines(mz, base_TH150, col = "steelblue")
```
Comment (sgibb) [inline R/estimateBaseline.R:134]:
```suggestion
##' base_TH15 <- estimateBaseline(mz, intensity,
##'                                method = "TopHat",
##'                                halfWindowSize = 15L)
##' lines(mz, base_TH15, col = "steelblue")
```

File: R/estimateBaseline.R [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+##' ## same as estimateBaselineSnip(mz, intensity, iterations = 100L)
+##' lines(mz, base_SNIP, col = "red")
+##'
+##' ## ----------------------------
+##' ## TopHat baseline
+##' base_TH75 <- estimateBaseline(mz, intensity,
+##'                               method = "TopHat",
+##'                               halfWindowSize = 75L)
+##' ## same as estimateBaselineTopHat(mz, intenstity, halfWindowSize = 75L)
+##' lines(mz, base_TH75, col = "blue")
```
Comment (sgibb) [inline R/estimateBaseline.R:129]:
```suggestion
##' base_TH25 <- estimateBaseline(mz, intensity,
##'                               method = "TopHat",
##'                               halfWindowSize = 25L)
##' ## same as estimateBaselineTopHat(mz, intenstity, halfWindowSize = 25L)
##' lines(mz, base_TH25, col = "blue")
```

File: R/estimateBaseline.R [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+##' intensity <- intensity + rnorm(nmz, mean = 0, sd = 1)
+##'
+##' plot(mz, intensity, type = "l")
+##'
+##' ## ----------------------------
+##' ## SNIP baseline
+##' base_SNIP <- estimateBaseline(mz, intensity,
+##'                               method = "SNIP",
+##'                               iterations = 100L)
+##' ## same as estimateBaselineSnip(mz, intensity, iterations = 100L)
```
Comment (sgibb) [inline R/estimateBaseline.R:120]:
```suggestion
##' ## same as estimateBaselineSnip(mz, intensity, iterations = 20L)
```

File: R/estimateBaseline.R [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+##' ## ----------------------------
+##' ## Median baseline
+##' base_med <- estimateBaseline(mz, intensity,
+##'                              method = "median")
+##' ## same as estimateBaselineMedian(mz, intensity)
+##' lines(mz, base_med, col = "orange")
+##'
+##' legend("topright", lwd = 1,
+##'         legend = c("SNIP", "TopHat (hws = 75)",
+##'                    "TopHat (hws = 150)",
```
Comment (sgibb) [inline R/estimateBaseline.R:152]:
```suggestion
##'         legend = c("SNIP", "TopHat (hws = 25)",
##'                    "TopHat (hws = 15)",
```

**Review by sgibb — APPROVED** [review]
Thanks @lgatto for integrating the baseline estimation algorithms!

I just changed some values for the iteration/half window sizes to demonstrate the difference in the simulated data.

**Comment by lgatto** [comment]
Thank you @sgibb!

**Pull request merged**